packages/web-app-files/src/components/SideBar/Details/FileDetails.vue

```diff
@@ -207,20 +237,37 @@ export default defineComponent({
     const preview = ref(undefined)
 
     const directLink = computed(() => {
-      return `${store.getters.configuration.server}files/spaces/personal/home${encodePath(
-        unref(resource).path
-      )}`
+      return !unref(isPublicLinkContext)
+        ? `${store.getters.configuration.server}files/spaces${encodePath(unref(resource).path)}`
+        : `${store.getters.configuration.server.replace(/\/+$/, '')}${unref(resource).downloadURL}`
     })
 
     const copyEosPathToClipboard = () => {
       copy(unref(resource).path)
       copiedEos.value = unref(copied)
       store.dispatch('showMessage', {
-        title: $gettext('EOS path copied'),
-        desc: $gettext('The EOS path has been copied to your clipboard.')
+        title: $gettext('FUSE path copied'),
+        desc: $gettext('The FUSE path has been copied to your clipboard.')
+      })
+    }
+
+    const copySambaPathToClipboard = () => {
+      copy(getSambaPath(unref(resource).path))
+      copiedSamba.value = unref(copied)
+      store.dispatch('showMessage', {
+        title: $gettext('Windows path copied'),
+        desc: $gettext('The Windows path has been copied to your clipboard.')
       })
     }
 
+    const getSambaPath = (path) => {
+      const pathComponents = path?.split('/').filter(Boolean)
```

Comment:
Can you please define the `pathMappings` inline here instead of having them in their own file? Completely CERN specific, so a dedicated file is probably more confusing than just having it inline here.

Reply:
Done, please let me know if it needs to be somewhere else. Also rebased to latest upstream.